### PR TITLE
OOP school library: add basic UI

### PR DESCRIPTION
**Pull request by Botlhale-Setou**
**Learning Outcomes Reached:**
- Build interactive console apps.
- Run a program using the command line.

### CODE REVIEW

File: app.rb [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+  def create_a_person()
+    puts 'Do you want to create a student(1) or a teacher(2)? [Enter a number 1 or 2]: '
+    cartegory = gets.chomp.to_i
+    case cartegory
+    when 1
+      print 'Enter the student name: '
+      name = gets.chomp
+      print 'Enter the student\'s age: '
+      age = gets.chomp
+      @people.push(Student.new('classroom', name, age))
```
Comment (ishimwezachee) [inline app.rb]: - [x]  I have found tow issue with this condition for your project to work as it was asked in the template video. 
- After correcting name and age from the user, you should also ask user if  a student has parent permission 
- After adding student age, the application is crushing, check if you are passing the parameters correctly 
-  Not necessary but I feel like if you start asking age, name respectively , will make your application more similar to the one provided from the template video.

File: app.rb [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+    when 2
+      print 'Enter the teacher name: '
+      name = gets.chomp
+      print 'Enter the teacher\'s age: '
+      age = gets.chomp
+      print 'Enter the teacher\'s specialization: '
+      specialization = gets.chomp
+      @people.push(Teacher.new(specialization, name, age))
+      puts "#{name.capitalize} was added as a teacher successfully"
+    end
```
Comment (ishimwezachee) [inline app.rb:58]: - [x]  After adding teacher's specialization, the application crushed as well as it happened above with student's age. please fix the issue by passing the argument correctly as it is in the template: 
image

**Review by ishimwezachee — CHANGES_REQUESTED** [review]
Hi @Botlhale-Setou , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
image


## Highlights
- Linters are passing   ✅
- Project is documented Professionally  ✅
- Project is running  ✅
## Required Changes ♻️

- [ ]  Because I could not create both teacher and student successfully (Person), it abided me to test other feature such as listing/creating persons, Rentals  etc.  Please fix those issues and make sure that your project is running exact way as it is provided in this [video template.](https://www.youtube.com/watch?v=vkkgrhD6aXQ&ab_channel=EmiliaAndrzejewska)  

_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app.rb (hunk 2 above)
Comment (Botlhale-Setou) [reply app.rb:58]: @ishimwezachee Thanks, I've updated the code and ensured all runs smoothly.

File: app.rb (hunk 1 above)
Comment (Botlhale-Setou) [reply app.rb]: @ishimwezachee Thanks, I've updated the code and ensured all runs smoothly.

File: app.rb [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+    cartegory = gets.chomp.to_i
+    case cartegory
+    when 1
+      print 'Enter the student name: '
+      name = gets.chomp
+      print 'Enter the student\'s age: '
+      age = gets.chomp
+      @people.push(Student.new('classroom', age, name))
+      puts "#{name.capitalize} was added as a student successfully"
+    when 2
```
Comment (BenMukebo) [inline app.rb]: - [ ] As requested by the previous reviewer, Kindly make sure After collecting the name and age of the student, you should also ask the user if the student has parental permission. See the screenshot bellow 👇🏻 

 image

### - And here is the previous reviewer requirement:
> 
>   - [x]  I have found tow issue with this condition for your project to work as it was asked in the template video. 
> - After correcting name and age from the user, you should also ask user if  a student has parent permission 
> - After adding student age, the application is crushing, check if you are passing the parameters correctly 
> -  Not necessary but I feel like if you start asking age, name respectively , will make your application more similar to the one provided from the template video.
> 
> _Originally posted by @ishimwezachee in https://github.com/Botlhale-Setou/school-library/pull/4#discussion_r1138891349_
>

File: app.rb [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+    end
+  end
+
+  # List all people method
+  def list_all_people
+    puts 'There are no people in the list. Kindly add at least one person' if @people.empty?
+    @people.each_with_index do |person, index|
+      puts "(#{index + 1}) [#{person.class}] => Id: #{person.id}, Name: #{person.name}, Age: #{person.age}"
+    end
+  end
```
Comment (BenMukebo) [inline app.rb:27]: - [ ] If you check the result, You can see that `(2) [Teacher] => Id: 80, Name: 845, Age: Doctor` You are display the age at the place of the name and the name at the place of the age, Kindly fix this 💯 👍🏻 


image

### - Here is the result:

image

**Review by BenMukebo — CHANGES_REQUESTED** [review]
Hi @Botlhale-Setou, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

## To Highlight :clap: 
- No linter errors. :heavy_check_mark:
- Gitflow is followed. :heavy_check_mark:
- PR has a good title and summary. ✔️


 ## Required Changes ♻️
_Check the comments under the review._

You have submitted a project, that Linters check are not running in the GitHub/action
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏 💻  
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please Tag me on (@BenMukebo) when you comment so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app.rb (hunk 3 above)
Comment (rloterh) [reply app.rb]: resolved! ✔️

File: app.rb (hunk 4 above)
Comment (rloterh) [reply app.rb:27]: resolved! ✔️

**Review by rloterh — APPROVED** [review]
## Status: Approved :heavy_check_mark:

Hi Botlhale,  

Your project is complete! 🏁 There is nothing else to say other than... it's time to merge it 🍾 Congratulations!  🎉💯🌟 

## Highlights 🎯 
💠 Interactive console app works as intended!

## Optional suggestions 🧰 
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you consider them as they can improve your code._

**You can also consider:**
- Kindly add appropriate instructions in your ReadMe under the _Getting Started_ section, on how to _run_ this app.
Implementing the above change(s) will make your ReadMe/ repository stand out even better! 😉

Cheers..Happy coding! 💻 🍷 🚀 

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**